dsl/src/main/scala/com.crobox.clickhouse/dsl/OperationalQuery.scala

```diff
@@ -130,51 +133,103 @@ trait OperationalQuery extends Query {
     newSelect
   }
 
-  def join[TargetTable <: Table](`type`: JoinQuery.JoinType, query: OperationalQuery): OperationalQuery = {
-    OperationalQuery(internalQuery.copy(join = Some(JoinQuery(`type`, InnerFromQuery(query)))))
-  }
+  def join[TargetTable <: Table](`type`: JoinQuery.JoinType,
+                                 query: OperationalQuery,
+                                 alias: Option[String]): OperationalQuery =
+    OperationalQuery(
+      internalQuery.copy(
+        join = Some(JoinQuery(`type`, InnerFromQuery(query), alias = alias.getOrElse(RandomStringGenerator.random())))
```

Comment:
The difficulty is that a single query might have many joins. Since a join might also involve a 'select query', it's hard to come up with a consistent naming convention. That said, it's up to the implementation of the API to provide an alias. If not explicitly set, a random one will be used.

I think we should keep it as is....